tests/helpers/test_instance_id.py

```diff
@@ -0,0 +1,26 @@
+"""Tests for instance ID helper."""
+from tests.async_mock import patch
+
+
+async def test_get_id_empty(hass, hass_storage):
+    """Get unique ID."""
+    uuid = await hass.helpers.instance_id.async_get()
+    assert uuid is not None
+    # Assert it's stored
+    assert hass_storage["core.uuid"]["data"]["uuid"] == uuid
+
+
+async def test_get_id_migrate(hass, hass_storage):
+    """Migrate existing file."""
+    with patch(
+        "homeassistant.util.json.load_json", return_value={"uuid": "1234"}
+    ), patch("os.path.isfile", return_value=True), patch("os.remove") as mock_remove:
+        uuid = await hass.helpers.instance_id.async_get()
+
+    assert uuid == "1234"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)